account_asset_management/report/account_asset_report_xls.py

```diff
@@ -0,0 +1,706 @@
+                title = prefix + " : " + _("Removed Assets")
+            else:
+                title = "DSP"
+        return title
+
+    def _report_title(self, ws, row_pos, ws_params, data, wiz):
+        return self._write_ws_title(ws, row_pos, ws_params)
+
+    def _empty_report(self, ws, row_pos, ws_params, data, wiz):
+        report = ws_params["report_type"]
+        if report == "acquisition":
+            suffix = _("New Acquisitions")
+        elif report == "active":
+            suffix = _("Active Assets")
+        else:
+            suffix = _("Removed Assets")
+        no_entries = _("No") + " " + suffix
+        ws.write_string(row_pos, 0, no_entries, FORMATS["format_left_bold"])
+
+    def _get_assets(self, wiz, data):
+        """ Add the selected assets, both grouped and ungrouped, to `data` """
+        dom = [
+            ("date_start", "<=", wiz.date_to),
+            "|",
+            ("date_remove", "=", False),
+            ("date_remove", ">=", wiz.date_from),
+        ]
+
+        parent_group = wiz.asset_group_id
+        if parent_group:
+
+            def _child_get(parent):
+                groups = [parent]
+                children = parent.child_ids
+                children = children.sorted(lambda r: r.code or r.name)
+                for child in children:
+                    if child in groups:
+                        raise UserError(
+                            _(
+                                "Inconsistent reporting structure."
+                                "\nPlease correct Asset Group '%s' (id %s)"
+                            )
+                            % (child.name, child.id)
+                        )
+                    groups.extend(_child_get(child))
+                return groups
+
+            groups = _child_get(parent_group)
+            dom.append(("group_ids", "in", [x.id for x in groups]))
+
+        if not wiz.draft:
+            dom.append(("state", "!=", "draft"))
+        assets = self.env["account.asset"].search(dom)
+        grouped_assets = {}
+        self._group_assets(assets, parent_group, grouped_assets)
+        data.update(
+            {
+                "assets": assets,
+                "grouped_assets": grouped_assets,
+            }
+        )
+
+    @staticmethod
+    def acquisition_filter(wiz, asset):
+        return asset.date_start >= wiz.date_from
+
+    @staticmethod
+    def active_filter(wiz, asset):
+        return True
+
+    @staticmethod
+    def removal_filter(wiz, asset):
+        return (
+            asset.date_remove
+            and asset.date_remove >= wiz.date_from
+            and asset.date_remove <= wiz.date_to
+        )
+
+    def _group_assets(self, assets, group, grouped_assets):
+        if group:
+            group_assets = assets.filtered(lambda r: group in r.group_ids)
+        else:
+            group_assets = assets
+        group_assets = group_assets.sorted(lambda r: (r.date_start or "", r.code))
```

Comment:
```suggestion
        group_assets = group_assets.sorted(lambda r: (r.date_start or "", r.code or "", r.name))
```
#1156

Reply:
Thanks! I've cherry picked the commit from the 13.0 branch.